django_cloud_deploy/cli/io.py

```diff
@@ -80,9 +151,39 @@ def ask(self, prompt=None):
         return input(self._replace_html_tags(prompt, sys.stdout.fileno()))
 
     def getpass(self, prompt=None):
-        "Prompt the user for a password and return the result."
+        """Prompt the user for a password and return the result."""
         return getpass.getpass(prompt)
 
+    @contextlib.contextmanager
+    def progressbar(self, expect_time: int, message: str):
+        """A context manager that shows a progress bar.
+
+        Output of the progress bar will be like the following:
+            Processing ██████████∙∙∙∙∙∙∙∙ Estimate time to finish: 7 seconds
+
+        If the task ends earlier than expected, the progress bar will directly
+        go to the end.
+
+        Args:
+            expect_time: How long the progress bar is going to run.
+                (In seconds).
+            message: A prefix of the progress bar showing what it is about.
+
+        Yields:
+            None
+        """
+
+        if os.isatty(sys.stdout.fileno()) and os.name == 'posix':
+            progress_bar = _ProgressBar(expect_time, message, posix=True)
```

Comment:
So the idea behind isatty is to determine whether this is an interactive console or not.

Does the progress bar work when writing to a text file e.g.
`$ django-cloud-deploy $(ARGS) >myfile.txt`

?

Reply:
Done. Only print estimate time to finish if not in an interactive console.